### ISSUE TITLE
Git Webhooks not working when "Applications in any namespace" is configured

### ISSUE THREAD

**Issue by mattstocum** [argoproj/argo-cd#13158]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When "Applications in any namespace" is configured, git push events do not trigger a repo refresh

**To Reproduce**

1. Install Argo CD from https://raw.githubusercontent.com/argoproj/argo-cd/v2.6.7/manifests/ha/install.yaml
2. Follow the instructions on https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/ to enable Applications in any namespace, with or without following recommended step for changing resource tracking methods.
3. Create an application linked to a git repository
4. Push a change to that repository

This happens regardless of whether the application is in the argocd namespace, or a separate namespace. With the additional namespaces disabled, updates work as expected.

**Expected behavior**

Argo CD should trigger an immediate refresh of any applications that use the repository.

**Version**

```shell
v2.7.0-rc1+0d0d2a9.dirty
and
v2.6.7
```

**Logs**

```
time="2023-04-07T18:50:45Z" level=info msg="Received push event repo: https://git.redacted/kubernetes/tkg-prod/mgr, revision: main, touchedHead: true"
time="2023-04-07T18:50:45Z" level=debug msg="https://git.redacted/kubernetes/tkg-prod/applications.git does not match (?i)^(http://|https://|[a-zA-Z0-9_\\.][a-zA-Z0-9_\\.-]{0,30}[a-zA-Z0-9_\\.\\$-]?@|ssh://([a-zA-Z0-9_\\.][a-zA-Z0-9_\\.-]{0,30}[a-zA-Z0-9_\\.\\$-]?@)?)git\\.lebow\\.drexel\\.edu(:[0-9]+|)[:/]kubernetes/tkg-prod/mgr(\\.git)?$"
time="2023-04-07T18:50:45Z" level=debug msg="https://git.redacted/kubernetes/tkg-prod/mgr.git uses repoURL https://git.redacted/kubernetes/tkg-prod/mgr"
time="2023-04-07T18:50:45Z" level=warning msg="Failed to refresh app 'applications-mgr' for controller reprocessing: error patching annotations in application \"applications-mgr\": the server could not find the requested resource (patch applications.argoproj.io applications-mgr)"
time="2023-04-07T18:50:45Z" level=debug msg="https://git.redacted/kubernetes/tkg-prod/mgr.git uses repoURL https://git.redacted/kubernetes/tkg-prod/mgr"
time="2023-04-07T18:50:45Z" level=warning msg="Failed to refresh app 'hello-world' for controller reprocessing: error patching annotations in application \"hello-world\": the server could not find the requested resource (patch applications.argoproj.io hello-world)"
```

**Comment by todaywasawesome** [comment]
Webhook looks up applications and probably uses the old interface.

**Comment by lukma99** [comment]
Any news on when the new PR will be merged so the problem is (hopefully) fixed soon?

**Comment by crenshaw-dev** [comment]
@lukma99 I requested changes on the PR, and the changes haven't been implemented yet. Do you have time to pick up the PR?

**Comment by speedfl** [comment]
The PR was opened the 10th of January. @crenshaw-dev do you maybe want me to take over this one ?

**Comment by Niksko** [comment]
I've raised a new PR to try and fix this. The claimed 'new PR that should fix this' #11930 is actually older than the original fix in #12386, so I figured it wasn't worth trying to integrate it, since it does basically the same thing.
